packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,636 @@
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 1.0, end: 0.0),
+      weight: 4 / 12,
+    ),
+    TweenSequenceItem<double>(
+      tween: ConstantTween<double>(0.0),
+      weight: 8 / 12,
+    ),
+  ]);
+  final _FlippableTweenSequence<double> _openOpacityTween =
+      _FlippableTweenSequence<double>(<TweenSequenceItem<double>>[
+    TweenSequenceItem<double>(
+      tween: ConstantTween<double>(0.0),
+      weight: 4 / 12,
+    ),
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.0, end: 1.0),
+      weight: 8 / 12,
+    ),
+  ]);
+
+  final GlobalKey _openBuilderKey = GlobalKey();
+
+  // Defines the position of the container on screen.
+  final RectTween _insetsTween = RectTween(end: Rect.zero);
+
+  // Defines the size of the container on screen.
+  final SizeTween _sizeTween = SizeTween();
+
+  AnimationStatus _lastAnimationStatus;
+  AnimationStatus _currentAnimationStatus;
+
+  @override
+  TickerFuture didPush() {
+    _takeMeasurements(navigatorContext: hideableKey.currentContext);
+
+    animation.addStatusListener((AnimationStatus status) {
+      _lastAnimationStatus = _currentAnimationStatus;
+      _currentAnimationStatus = status;
+      switch (status) {
+        case AnimationStatus.dismissed:
+          hideableKey.currentState
+            ..placeholderSize = null
+            ..isVisible = true;
+          break;
+        case AnimationStatus.completed:
+          hideableKey.currentState
+            ..placeholderSize = null
+            ..isVisible = false;
+          break;
+        case AnimationStatus.forward:
+        case AnimationStatus.reverse:
+          break;
+      }
+    });
+
+    return super.didPush();
+  }
+
+  @override
+  bool didPop(void result) {
+    _takeMeasurements(
+      navigatorContext: subtreeContext,
+      delayForSourceRoute: true,
+    );
+    return super.didPop(result);
+  }
+
+  void _takeMeasurements({
+    BuildContext navigatorContext,
+    bool delayForSourceRoute = false,
+  }) {
+    final RenderBox navigator =
+        Navigator.of(navigatorContext).context.findRenderObject();
+    final Size navSize = _getSize(navigator);
+    _sizeTween.end = navSize;
+
+    void takeMeasurementsInSourceRoute([Duration _]) {
+      final Rect srcRect = _getRect(hideableKey, navigator);
+      _sizeTween.begin = srcRect.size;
+      _insetsTween.begin = Rect.fromLTRB(
+        srcRect.left,
+        srcRect.top,
+        navSize.width - srcRect.right,
+        navSize.height - srcRect.bottom,
+      );
+      hideableKey.currentState.placeholderSize = _sizeTween.begin;
+    }
+
+    if (delayForSourceRoute) {
+      SchedulerBinding.instance
+          .addPostFrameCallback(takeMeasurementsInSourceRoute);
+    } else {
+      takeMeasurementsInSourceRoute();
+    }
+  }
+
+  Size _getSize(RenderBox render) {
+    assert(render != null && render.hasSize);
+    return render.size;
+  }
+
+  Rect _getRect(GlobalKey key, RenderBox ancestor) {
```

Comment:
This might have a slightly more informative name. We're getting the bounds of the renderer identified by key in its ancestor's coordinate system. Of course I'm not sure how to make a short name out of that. Maybe a comment.